src/main/backend/setup.ts

```diff
@@ -267,17 +269,21 @@ const setupOwnedBackend = async (
     });
     const port = await getValidPort();
 
+    const currentExecutableDir = path.dirname(app.getAppPath());
+    const isPortable = await checkFileExists(path.join(currentExecutableDir, 'portable'));
+    const rootDir = isPortable ? currentExecutableDir : app.getPath('userData');
```

Comment:
I think the logic for getting the root dir should be moved out of `backend/setup.ts` completely. It could be passed in via `getRootDir: () => Promise<string>` parameter, like `hasNvidia`.